kernels/TIndexKernel.cpp

```diff
@@ -118,6 +118,15 @@ void TIndexKernel::addSubSwitches(ProgramArgs& args,
             "Write absolute rather than relative file paths", m_absPath);
         args.add("stdin,s", "Read filespec pattern from standard input",
             m_usestdin);
+        args.add("threads", "Number of threads to use for file boundary creation",
+            m_threads, 1);
+        args.add("smooth", "Smooth boundary output", m_doSmooth, true);
+        args.add("threshold", "Required cell density, to be used in internal "
+            "hexbin filter (non-fast boundary)", m_density, 15);
+        args.add("edge_length", "cell edge length to be used in internal hexbin "
+            "filter (non-fast boundary)", m_edgeLength);
+        args.add("sample_size", "Sample size for auto-edge length calculation in "
```

Comment:
Do we have to expose this to users? They can do `--filters.hexbin.sample_size` themselves if they want.